e2e/utils/global-setup.ts

```diff
@@ -57,8 +57,19 @@ export default async function (globalConfig: Config.ConfigGlobals) {
     !existsSync('./build')
   ) {
     console.log('Publishing packages to local registry');
-    execSync('pnpm nx-release --local major', {
-      stdio: isVerbose ? 'inherit' : 'ignore',
-    });
+    try {
+      execSync('pnpm nx-release --local major', {
+        stdio: isVerbose ? 'inherit' : 'pipe',
+      });
+    } catch (err) {
+      if (err?.stdout) {
+        process.stdout.write(err.stdout.toString());
+      }
+      if (err?.stderr) {
+        process.stderr.write(err.stderr.toString());
+      }
+      console.log(err);
+      throw err;
+    }
```

Comment:
Your solution would have all of the `stdout` followed by all of the `stderr` instead of interlaced which would be normally how it is printed. Below would be better

```suggestion
    await new Promise<void>((res) => {
      const publishProcess = exec('pnpm nx-release --local major');

      let logs = Buffer.from('');
      if (isVerbose) {
        publishProcess.stdout.pipe(process.stdout);
        publishProcess.stderr.pipe(process.stderr);
      } else {
        publishProcess.stdout.on('data', (data) => (logs += data));
        publishProcess.stderr.on('data', (data) => (logs += data));
      }
      publishProcess.on('exit', (code) => {
        if (!isVerbose && code > 0) {
          console.log(logs.toString());
        }
        res();
      });
```

Reply:
changed